hw/nfc-nci.c

```diff
@@ -651,41 +651,19 @@ init_process_oid_rf_discover_select_cmd(const union nci_packet* cmd,
 struct nfc_deactivate_ntf_param {
     uint8_t type;
     uint8_t reason;
+    struct nfc_device* nfc;
 };
 
-static ssize_t
-nfc_delivery_deactivate_cmd_cb(void* data, union nci_packet* pkt)
-{
-    struct nfc_deactivate_ntf_param* param;
-    ssize_t res;
-
-    assert(data);
-
-    param = (struct nfc_deactivate_ntf_param*)data;
-    res = nfc_create_deactivate_ntf(param->type, param->reason, pkt);
-
-    qemu_free(param);
-
-    return res;
-}
-
-static size_t
-init_process_oid_rf_deactivate_cmd(const union nci_packet* cmd,
-                                   struct nfc_device* nfc,
-                                   union nci_packet* rsp,
-                                   struct nfc_delivery_cb* cb)
+static void
+nfc_rf_state_deactivate_transition(enum nci_rf_deactivation_type type,
```

Comment:
Maybe just call this function |perform_deactivate_transition|. All the other functions with 'nfc_rc_state_' prefix are located in nfc-rf.{c,h}.